### DISCUSSION TITLE
How to set AI model temperature?

### DISCUSSION THREAD

**Discussion by benmyles** [zed-industries/zed#17715] [Feature Requests]
Is there any way to set the temperature of the ai model?

I expected something like this to work:
```json
    "anthropic": {
      "version": "1",
      "available_models": [
        {
          "name": "claude-3-5-sonnet-20240620",
          "display_name": "claude-3-5-sonnet-8k-t0",
          "temperature": 0.0,
          "max_tokens": 195000,
          "max_output_tokens": 8000,
          ...
```

But it doesn't: `temperature` just gets ignored. If it's not possible, is it on the todo list already? The temperature makes a significant difference to the outputs.

**Comment by benmyles** [comment]
I did find [Pass temperature to Anthropic #17509](https://github.com/zed-industries/zed/pull/17509/files) so it looks like it's hard-coded at the moment. Hopefully there are plans to make it configurable.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
